File: Python/PCA9536_R21.py
# Distributed with a free-will license.
# Use it any way you want, profit or free, provided it fits in the licenses of its associated works.
# PCA9536_R21
# This code is designed to work with the PCA9536_I2CR_R21 I2C Mini Module available from ControlEverything.com.
# https://www.controleverything.com/products

import smbus
import time

# Get I2C bus
bus = smbus.SMBus(1)

# PCA9536_R21 address, 0x41(65)
# Select configuration register, 0x03(03)
#		0x00(00)	Set all pins as OUTPUT
bus.write_byte_data(0x41, 0x03, 0x00)
time.sleep(0.5)

# PCA9536_R21 address, 0x41(65)
# Select output port register, 0x01(01)
#		0x01(01)	Set Pin-1 HIGH
bus.write_byte_data(0x41, 0x01, 0x01)
time.sleep(1);

# Output to screen
print "Pin-1 state is : HIGH"

# PCA9536_R21 address, 0x41(65)
# Select output port register, 0x01(01)
#		0x00(00)	Set Pin-1 LOW
bus.write_byte_data(0x41, 0x01, 0x00)
time.sleep(1);

# Output to screen
print "Pin-1 state is : LOW"

# PCA9536_R21 address, 0x41(65)
# Select output port register, 0x01(01)
#		0x02(02)	Set Pin-2 HIGH
bus.write_byte_data(0x41, 0x01, 0x02)
time.sleep(1);

# Output to screen
print "Pin-2 state is : HIGH"

# PCA9536_R21 address, 0x41(65)
# Select output port register, 0x01(01)
#		0x00(00)	Set Pin-2 LOW
bus.write_byte_data(0x41, 0x01, 0x00)
time.sleep(1);

# Output to screen
print "Pin-2 state is : LOW"

File: C/PCA9536_R21.c
// Distributed with a free-will license.
// Use it any way you want, profit or free, provided it fits in the licenses of its associated works.
// PCA9536_R21
// This code is designed to work with the PCA9536_I2CR_R21 I2C Mini Module available from ControlEverything.com.
// https://www.controleverything.com/products

#include <stdio.h>
#include <stdlib.h>
#include <linux/i2c-dev.h>
#include <sys/ioctl.h>
#include <fcntl.h>

void main() 
{
	// Create I2C bus
	int file;
	char *bus = "/dev/i2c-1";
	if((file = open(bus, O_RDWR)) < 0) 
	{
		printf("Failed to open the bus. \n");
		exit(1);
	}
	// Get I2C device, PCA9536_R11 I2C address is 0x41(65)
	ioctl(file, I2C_SLAVE, 0x41);

	// Select configuration register(0x03)
	// Set all pins as OUTPUT(0x00)
	char config[2] = {0};
	config[0] = 0x03;
	config[1] = 0x00;
	write(file, config, 2);
	sleep(1);

	// Select output port register(0x01)
	// Set pin-1 as HIGH(0x01)
	char config[2] = {0};
	config[0] = 0x01;
	config[1] = 0x01;
	write(file, config, 2);
	sleep(1);
	
	// Output to screen
	printf("Pin-1 state is : HIGH");
	
	// Select output port register(0x01)
	// Set pin-1 as LOW(0x00)
	char config[2] = {0};
	config[0] = 0x01;
	config[1] = 0x00;
	write(file, config, 2);
	sleep(1);
	
	// Output to screen
	printf("Pin-1 state is : LOW");
	
	// Select output port register(0x01)
	// Set pin-2 as HIGH(0x02)
	char config[2] = {0};
	config[0] = 0x01;
	config[1] = 0x02;
	write(file, config, 2);
	sleep(1);
	
	// Output to screen
	printf("Pin-2 state is : HIGH");
	
	// Select output port register(0x01)
	// Set pin-2 as LOW(0x00)
	char config[2] = {0};
	config[0] = 0x01;
	config[1] = 0x00;
	write(file, config, 2);
	sleep(1);
	
	// Output to screen
	printf("Pin-2 state is : LOW");
}
